### ISSUE TITLE
couldn't support flutter beta2?

### ISSUE THREAD

**Issue by arden** [mubarak1361/flutter_todo#1]
this example couldn't support flutter beta2?

**Comment by mubarak1361** [comment]
yes, it support.

**Comment by arden** [comment]
there some errors.

```
Launching lib/main.dart on iPhone X in debug mode...
Running Xcode clean...                                       3.0s
Starting Xcode build...                                          
 ├─Assembling Flutter resources...                    6.3s
 └─Compiling, linking and signing...                 42.8s
Xcode build done                                            56.4s
Directory: '/Users/weibo/Library/Developer/CoreSimulator/Devices/A9A209D8-F7DB-43C5-BCAC-057B860D49F4/data/Containers/Data/Application/CCCE4E08-487C-4595-9755-AEEA46278AB0/Documents'
[VERBOSE-2:dart_error.cc(16)] Unhandled exception:
NoSuchMethodError: The method 'insert' was called on null.
Receiver: null
Tried calling: insert(0, Instance of 'Category')
#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:46:5)
#1      TodoListState._getCategoryList.<anonymous closure> (package:flutter_todo/ui/todo_list.dart:48:18)
#2      _RootZone.runUnary (dart:async/zone.dart:1381:54)
#3      _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
#4      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:633:45)
#5      Future._propagateToListeners (dart:async/future_impl.dart:662:32)
#6      Future._complete (dart:async/future_impl.dart:467:7)
#7      _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
#8      DatabaseHelper.closeDatabase (package:flutter_todo/db/database.dart)
<asynchronous suspension>
#9      CategoryProvider._close (package:flutter_todo/db/category_provider.dart:18:43)
#10     CategoryProvider.getAllCategory.<an<…>
Directory: '/Users/weibo/Library/Developer/CoreSimulator/Devices/A9A209D8-F7DB-43C5-BCAC-057B860D49F4/data/Containers/Data/Application/CCCE4E08-487C-4595-9755-AEEA46278AB0/Documents'
Syncing files to device iPhone X...                          5.1s

🔥  To hot reload your app on the fly, press "r". To restart the app entirely, press "R".
An Observatory debugger and profiler on iPhone X is available at: http://127.0.0.1:8101/
For a more detailed help message, press "h". To quit, press "q".
Directory: '/Users/weibo/Library/Developer/CoreSimulator/Devices/A9A209D8-F7DB-43C5-BCAC-057B860D49F4/data/Containers/Data/Application/CCCE4E08-487C-4595-9755-AEEA46278AB0/Documents'
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following NoSuchMethodError was thrown building NewTodo(dirty, state: NewTodoState#efad6):
The getter 'isNotEmpty' was called on null.
Receiver: null
Tried calling: isNotEmpty

When the exception was thrown, this was the stack:
#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:46:5)
#1      NewTodoState.build (package:flutter_todo/ui/new_todo.dart:100:31)
#2      StatefulElement.build (package:flutter/src/widgets/framework.dart:3713:27)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3625:15)
#4      Element.rebuild (package:flutter/src/widgets/framework.dart:3478:5)
#5      BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2225:33)
#6      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:621:20)
#7      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:208:5)
#8      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:990:15)
#9      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:930:9)
#10     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:842:5)
#11     _invoke (dart:ui/hooks.dart:120:13)
#12     _drawFrame (dart:ui/hooks.dart:109:3)
════════════════════════════════════════════════════════════════════════════════════════════════════
```

**Comment by mubarak1361** [comment]
now, it's fixed. please check the  commit ` bacd183`
